base/collection_xattr.go

```diff
@@ -630,3 +626,27 @@ func (c *Collection) DeleteUserXattr(k string, xattrKey string) (uint64, error)
 	}
 	return uint64(result.Cas()), nil
 }
+
+// initializeMutateInSpec will append macro expansions defined in MutateInOptions to the provided
+// gocb MutateInSpec.
+func (c *Collection) appendMacroExpansions(mutateInSpec []gocb.MutateInSpec, opts *sgbucket.MutateInOptions) []gocb.MutateInSpec {
```

Comment:
I'm loosely inclined to make this not be a method on `Collection` because it doesn't have a dependency on `Collection`